Fix file.line diff formatting.

gvengel opened the pull request:
### What does this PR do?

Fixes formatting issue in file.line's changes diff output.

By default, splitlines() strips off line endings, which mangles the unified_diff output. Setting keepends=True resolves this issue.

Upstream versions have replaced splitlines() with _splitlines_preserving_trailing_newline(). This can be switched to splitlines(True) which does preserve trailing newlines.

### What issues does this PR fix or reference?

Fixes #41474

### Previous Behavior

Changes diff displays compressed output on a single line.

```
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,4 +1,5 @@
                   egg bacon+sausage spam
```

### New Behavior

Changes diff displays multiline output.

     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -1,3 +1,4 @@
                   egg
                   bacon
                  +sausage
                   spam

### Tests written?

No

Commit:
Always notify ryan-lane when changes occur on boto files

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
git.detached: Fix traceback when rev is a SHA and is not present locally

This catches the (expected) CommandExecutionError when git.describe is
run on a commit that doesn't exist.

This also renames the remote_ref_type variable to remote_rev_type. This
change was made for the 2017.7 branch, but this fix is being applied to
2016.11, so by making the rename here as well we will avoid a potential
bug from creeping in on a merge-forward.

terminalmage committed:
Fix missed deprecation

This removes an argument that was slated for deprecation in the Hydrogen
release.

basepi committed:
Add basepi to userBlacklist for mention bot

Commit:
Merge pull request #43179 from terminalmage/old-deprecation

Fix missed deprecation

Commit:
Fix apache.config with multiple statement
At this moment when you post more than one statement in config only last
is used. Also file is rewrited multiple times until last statement is
written.
Example:
salt '*' apache.config /etc/httpd/conf.d/ports.conf config="[{'Listen': '8080'}, {'Proxy': "Something"}]"
Ends only with
   Proxy Something
and ignore Listen 8080,
This patch fix this issue.

Commit:
Merge pull request #43178 from terminalmage/issue43143

git.detached: Fix traceback when rev is a SHA and is not present locally

gtmanfred committed:
Pin request install to version

This fails on centos 6 because its node is too old to support the version of
hawk bumped here https://github.com/request/request/pull/2751, we can still
test the functionality.  This will pull from github, and install a specific tag
version, and we still do the uninstall using the github path.

This should be more stable.

corywright committed:
Add `disk.format` alias for `disk.format_`

lomeroe committed:
correcting bad format statement in search for policy to be disabled

Commit:
Merge pull request #43196 from gtmanfred/2016.11

Pin request install to version for npm tests

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

Commit:
Merge pull request #43199 from corywright/disk-format-alias

Add `disk.format` alias for `disk.format_`

garethgreenaway committed:
Reverting this change due to it breaking other uses.

Commit:
Merge pull request #42663 from StreetHawkInc/fix_git_tag_check

Check remote tags before deciding to do a fetch #42329

Commit:
Merge pull request #43103 from aogier/43101-genesis-bootstrap

genesis.bootstrap deboostrap fix

Commit:
Always notify tkwilliams when changes occur on boto files

Commit:
Merge pull request #43202 from garethgreenaway/42642_2016_11_augeas_module_revert_fix

Reverting previous augeas module changes

lomeroe committed:
verify that files exist before trying to remove them, win_file.remove
raises an exception if the file does not exist

Commit:
Merge remote-tracking branch 'upstream/2016.11' into fix-apache-config-multi-entity

Commit:
Merge pull request #43154 from lomeroe/bp-43116-2016.11

Backport #43116 to 2016.11

twangboy committed:
Fix pkg.install

I jacked this up previously
convert cmd to a list
use cmd.extend

Commit:
Merge pull request #43191 from viktorkrivak/fix-apache-config-multi-entity

Fix apache.config with multiple statement

twangboy committed:
Use regex to detect salt-minion install

twangboy committed:
Fix spelling

Commit:
Update release branch section with a few more details

twangboy committed:
Add underscore to regex search

Commit:
Merge branch '2016.11' into issue-35840-fix-preserve-minion-cache-2016.11

twangboy committed:
Fix regex statement, add `.`

Commit:
Merge pull request #43228 from twangboy/win_fix_pkg.install

Win fix pkg.install

Commit:
Merge branch '2016.11' into issue-35840-fix-preserve-minion-cache-2016.11

twangboy committed:
Fix formatting issue, spaces surrounding +

Commit:
Merge pull request #43271 from twangboy/win_fix_pkg.install

Fix minor formatting issue

terminalmage committed:
Use six.integer_types instead of int

This catches longs on PY2.

Commit:
Merge pull request #43274 from terminalmage/fix-int-types

Use six.integer_types instead of int

Commit:
Add CODEOWNERS file

lordcirth committed:
file.py docs: specify absolute paths

lordcirth committed:
file.py docs: correct group and mode

Commit:
Merge pull request #43290 from lordcirth/fix-file-path-docs

Clarify file.py docs

lordcirth committed:
cron docs: Remind user to use quotes for special strings

Otherwise @ signs are stripped.
https://github.com/saltstack/salt/issues/2896

Commit:
Merge pull request #43312 from lordcirth/fix-cron-docs

cron docs: Remind user to use quotes for special strings

Commit:
Merge pull request #43277 from rallytime/owners-file

Add CODEOWNERS file

doesitblend committed:
Update Salt Mine documentation to show that the mine_interval option is configured in minutes.

doesitblend committed:
Fix phrasing for mine_interval description

johnj committed:
Fix #43295, better handling of consul initialization issues

damon-atkins committed:
Docs are wrong cache_dir (bool) and cache_file (str) cannot be passed on the cli (#2)

damon-atkins committed:
Docs are wrong cache_dir (bool) and cache_file (str) cannot be passed on the cli (#2)

damon-atkins committed:
Merge branch '2016.11' into 2016.11

damon-atkins committed:
Update win_pkg.py

damon-atkins committed:
Update win_pkg.py

gtmanfred committed:
never-download got readded

https://github.com/pypa/virtualenv/pull/840

gtmanfred committed:
ipaddr_start ipaddr_end for el7

Commit:
Merge pull request #43247 from rallytime/mentionbot-backports

Back-port various mention bot settings to 2016.11

Commit:
Merge pull request #43359 from gtmanfred/ipaddr

ipaddr_start ipaddr_end for el7

gtmanfred committed:
fix test

terminalmage committed:
Move --showduplicates before repository-packages

As of Fedora 26, dnf interprets arguments after "repository-packages" as
arguments to the repository-package subcommand, which breaks
pkg.list_repo_pkgs on that Fedora release. Moving this argument earlier
in the command allows pkg.list_repo_pkgs to work.

terminalmage committed:
Fix KeyError in yumpkg configparser code on Python 3

It looks like the configparser in Python 3 does not insert the
`__name__` key in each section. Popping it without a default causes a
KeyError on Python 3. This commit fixes that KeyError.

Commit:
Merge pull request #43244 from rallytime/release-branch-clarifications

Update release branch section with a few more details

Commit:
Merge pull request #43360 from terminalmage/sj-496

Fix failing tests in Fedora

Commit:
Merge pull request #42903 from junovitch/issue-35840-fix-preserve-minion-cache-2016.11

Fix 'preserve_minion_cache: True' functionality (fixes #35840)

Commit:
Merge pull request #43361 from rallytime/bp-43329

Back-port #43329 to 2016.11

damon-atkins committed:
Update doco

damon-atkins committed:
Update doco

brejoc committed:
Catching error when PIDfile cannot be deleted

Usually the PIDfile is locate in /run. If Salt is not started with root
permissions, it is not able to delete the PIDfile in /run. It should
be safe to just log and ignore this error, since Salt overwrites the
PIDfile on the next start.

brejoc committed:
Removed unused format argument

Commit:
Merge pull request #43333 from damon-atkins/2016.11

Docs are wrong cache_dir (bool) and cache_file (str) cannot be passed as params + 1 bug

Commit:
Merge branch '2016.11' into 43086-no-member

Commit:
Merge pull request #43105 from aogier/43086-no-member

groupadd module: string does not have attribute 'extend', plus homogeneous `cmd` parm building

doesitblend committed:
Fix mine_interval phrasing in default file

Commit:
Lint: Remove trailing whitespace

Commit:
better qemu_static parameter mangle in deboostrap management, tests

Commit:
Add handler.messages back in for test comparison

Commit:
Merge pull request #43325 from doesitblend/salt-mine-doc-fix

mine_interval option is minutes not seconds

Commit:
Merge pull request #43356 from gtmanfred/2016.11

never-download got readded

Commit:
only check if static_qemu is_executable()

mattLLVW committed:
Fix env_order in state.py

Fixes #42165

Commit:
forgot to mock the proper one

gtmanfred committed:
make cache dirs when spm starts

Commit:
Merge pull request #43390 from aogier/43387-genesis-qemu

better qemu_static parameter mangle in deboostrap management, tests

Commit:
Add 2016.11.8 release notes

Commit:
Merge pull request #43434 from rallytime/2016.11.8-release-notes

Add 2016.11.8 release notes

Commit:
Merge pull request #43432 from rallytime/bp-43419

Back-port #43419 to 2016.11

meaksh committed:
Use $HOME to get the user home directory instead using '~' char

garethgreenaway committed:
Adding a small check to ensure we do not continue to populate kwargs with __pub_ items from the kwargs item.

Commit:
Bump deprecation warning for boto_vpc.describe_route_table

This deprecation warning needs to be bumped out to Neon instead
of Oxygen.

See Issue #43223 for more details.

Commit:
Merge pull request #43445 from rallytime/bump-deprecation-warning

Bump deprecation warning for boto_vpc.describe_route_table

Commit:
Merge pull request #43441 from meaksh/2016.11-salt-bash-completion-fix

Use $HOME to get the user home directory instead using '~' char

Commit:
Merge branch '2016.11' into 43386_2016_11_schedule_kwargs_pub

Commit:
Add Neon to version list

Follow up to #43445

terminalmage committed:
Fix missing PER_REMOTE_ONLY in cache.clear_git_lock runner

This allows for the runner to clear the git lock if any of the
per-remote-only params was present in the configuration.

This also adds PER_REMOTE_ONLY attributes to the winrepo runner and
git_pillar external pillar, and fixes all refs to GitBase's init_remotes
function to include PER_REMOTE_ONLY arguments. This will insulate us
against future headaches if either git_pillar or winrepo gain extra
per-remote-only params, as we will only need to make the change to the
attribute in their respective modules.

Commit:
Merge pull request #43456 from rallytime/43445_follow_up

Add Neon to version list

Commit:
Merge pull request #43442 from garethgreenaway/43386_2016_11_schedule_kwargs_pub

 [2016.11] Fixes to scheduler __pub values in kwargs

Commit:
Merge pull request #43366 from brejoc/2016.11.pidfile-fix

Catching error when PIDfile cannot be deleted

twangboy committed:
Add /norestart switch to vcredist install

twangboy committed:
Handle ErrorCodes returned by VCRedist installer

twangboy committed:
Change all comment markers to '#'

Commit:
Merge pull request #43461 from twangboy/win_norestart

Add `/norestart` switch to vcredist install

garethgreenaway committed:
Merge pull request #43508 from rallytime/bp-43333

Back-port  #43333 to 2016.11.8

twangboy committed:
Add posix=False to call to salt.utils.shlex_split

twangboy committed:
Fix logic in `/etc/paths.d/salt` detection

vutny committed:
cloud.action: list_nodes_min returns all instances

akissa committed:
Fix CSR not recreated if key changes

Commit:
Merge pull request #43550 from twangboy/osx_fix_preinstall_2016.11.8

Fix preinstall script on OSX for 2016.11.8

Commit:
Merge pull request #43572 from vutny/fix-salt-cloud-list-min-instance-set

cloud.action: list_nodes_min returns all EC2 instances

Commit:
Merge branch '2016.11.8' into '2016.11'

No conflicts.

Commit:
Merge pull request #43591 from rallytime/merge-2016.11

[2016.11] Merge forward from 2016.11.8 to 2016.11

defanator committed:
Several fixes for RDS DB parameter group management

In particular:

 - it is now possible to manage all the parameters in a group,
   without limiting to MaxRecords=100 (thanks to pagination);

 - update_parameter_group() now composes valid JSON payload,
   automatically substitutes boolean values to 'on' / 'off' strings;

 - parameter_present() now shows actual error message produced by
   ModifyDBParameterGroup API call.

brejoc committed:
Added tests for pid-file deletion in DaemonMixIn

This is a follow up on this PR:
https://github.com/saltstack/salt/pull/43366

Since we can get an OSError durin PIDfile deletion with non-root users, it
would make sense to also test for this. So here are the two test cases.
One with an OSError and the other one without.

moio committed:
multiprocessing minion option: documentation fixes

brejoc committed:
Fixed several issues with the test

* Removed redundant import.
* No longer inheriting from LogSettingsParserTests.
* Replaced test class description with somethin meaninful.
* Fixed identation.

I've also moved the class to the bottom, since all the classes inheriting
from LogSettingsParserTests are in the block above.

brejoc committed:
Fixed linting

Fix for "String format call with un-indexed curly braces".

Commit:
Merge pull request #43646 from brejoc/2016.11.4-pidfile-tests

Added tests for pid-file deletion in DaemonMixIn

Commit:
Handle VPC/Subnet ID not found errors in boto_vpc module

If a VPC or Subnet ID is not found when calling functions that are
supposed to be checking for vpc/subnet ID existence, the return
should be consistent by returning booleans/None instead of returning
the NotFound error from AWS.

The surrounding code blocks indicate that this is expected as well
as unit test assertions. The moto library had a bug in it where
it wasn't raising "x.NotFound" errors when it should have been. The
latest version of moto has fixed this bug, causing our tests to fail
since the boto_vpc module is not handling the "x.NotFound" errors
separately from the generic BotoServerErrors.

This fixes the test failures in the branch tests that were caused by
upgrading the moto version to the latest release.

VertigoRay committed:
Salt Repo has Deb 9 and 8

Commit:
Merge pull request #43661 from moio/2016.11-multiprocessing-doc-fix

multiprocessing minion option: documentation fixes (2016.11)

Commit:
Merge pull request #43534 from twangboy/win_fix_pkg.install_2016.11

Fixes removal of double-quotes by shlex_split in winrepo for 2016.11

terminalmage committed:
Fix RST headers for runners (2016.11 branch)

To conform with the rest of the rst files for runner docs, they should
only contain the module name.

Commit:
Merge pull request #43677 from terminalmage/runners-docs-2016.11

Fix RST headers for runners (2016.11 branch)

Commit:
Merge pull request #43673 from rallytime/bp-43652

Back-port #43652 to 2016.11

Commit:
Merge pull request #43672 from rallytime/bp-43415

Back-port #43415 to 2016.11

Commit:
Merge pull request #43575 from akissa/fix-csr-not-recreated-if-key-changes

Fix CSR not recreated if key changes

Commit:
Merge pull request #43648 from rallytime/handle-boto-vpc-errors

Handle VPC/Subnet ID not found errors in boto_vpc module

terminalmage committed:
Add missing support for use/use_in requisites to state.sls_id

Because requisite resolution doesn't happen until we run call_high, and
state.sls_id doesn't run call_high (but rather calls a single low chunk
from the compiled low chunks), the use/use_in requisites are ignored.
This fixes that oversight by resolving requisites in state.sls_id before
we compile the high data.

Commit:
Revise "Contributing" docs: merge-forwards/release branches explained!

Fixes #43650

The merge-forward process needs a more prominent position and explanation
in the contributing documentation. This change attempts to explain this
process a little more fully and incorporates some changes to how we are
handling "main" release branches, "dot" release branches, and "develop"
in a more complete context with merge-forwards.

terminalmage committed:
Fix failing unit test

terminalmage committed:
Allow docker_events engine to work with newer docker-py

The Client attribute was renamed to APIClient in docker-py 2.0

Commit:
Add GPG Verification section to Contributing Docs

When we enable GPG Verification for pull request reviews, we should
make sure there is information available in our Contributing docs
about how to sign commits.

Commit:
Merge pull request #43727 from rallytime/fix-43650

Revise "Contributing" docs: merge-forwards/release branches explained!

Commit:
Merge branch '2016.11' into issue43373

Commit:
Merge pull request #43458 from terminalmage/issue42082

Fix missing PER_REMOTE_ONLY in cache.clear_git_lock runner

Commit:
Merge pull request #43733 from terminalmage/issue43729

Allow docker_events engine to work with newer docker-py

Commit:
Merge branch '2016.11' into gpg-verification

Commit:
Merge pull request #43747 from rallytime/gpg-verification

Add GPG Verification section to Contributing Docs

vutny committed:
Fix Pylint deprecated option warnings

gtmanfred committed:
dont print Minion not responding with quiet

garethgreenaway committed:
Merge branch '2016.11' into issue43373

garethgreenaway committed:
Merge pull request #43772 from gtmanfred/2016.11

dont print Minion not responding with quiet

garethgreenaway committed:
Merge branch '2016.11' into fix-pylint-deprecation-warnings

Ch3LL committed:
[2016.11] Bump latest and previous versions

Ch3LL committed:
Add Security Notes to 2016.3.8 Release Notes

Ch3LL committed:
Add Security Notes to 2016.11.8 Release Notes

Commit:
Merge pull request #43768 from vutny/fix-pylint-deprecation-warnings

Fix Pylint deprecated option warnings

terminalmage committed:
cmdmod: Don't list-ify string commands on Windows

See https://github.com/saltstack/salt/issues/43522.

garethgreenaway committed:
Merge pull request #43807 from terminalmage/issue43522

cmdmod: Don't list-ify string commands on Windows

Commit:
add -n with netstat so we don't resolve

Commit:
Merge pull request #43707 from terminalmage/issue43373

Add missing support for use/use_in requisites to state.sls_id

terminalmage committed:
Only join cmd if it's not a string

See https://github.com/saltstack/salt/pull/43807#discussion_r142009644

UtahDave committed:
Update SaltConf banner per Rhett's request

Commit:
Merge pull request #43869 from terminalmage/issue43522

Only join cmd if it's not a string

garethgreenaway committed:
Merge branch '2016.11' into 2016.11local

Commit:
Merge pull request #43884 from UtahDave/2016.11local

Update SaltConf banner per Rhett's request

dereckson committed:
Fix typo in salt-cloud scaleway documentation

s/scalewa/scaleway

Commit:
Merge pull request #43916 from dereckson/fix-typo-cloud-scaleway

Fix typo in salt-cloud scaleway documentation

Commit:
Merge pull request #43888 from rallytime/bp-43841

Back-port #43841 to 2016.11

meaksh committed:
Enable '--with-salt-version' parameter for setup.py script

cro committed:
Release notes blurb for change to bindpw requirements

cro committed:
Require that bindpw be non-empty if auth.ldap.anonymous=False

bobrik committed:
Report built-in modiles in kmod.available, fixes #43945

rossengeorgiev committed:
fix zenoss state module not respecting test=true

terminalmage committed:
Fix grains.has_value when value is False

This function returns a boolean based on the boolean result of the value
returned when looking up the key. However, this means that if a key
exists and has a value of 0, an empty string, or False/None,
grains.has_value will incorrectly return False. This fixes that by only
returning False when the key is not present.

Commit:
Merge pull request #43973 from terminalmage/fix-grains.has_value

Fix grains.has_value when value is False

oeuftete committed:
Fix manage.present to show lost minions

Fixes #38367 and and #43936.

garethgreenaway committed:
Merge pull request #43977 from Ch3LL/3.8_sec

Add Security Notes to 2016.3.8 Release Notes

garethgreenaway committed:
Merge pull request #43976 from Ch3LL/11.8_sec

Add Security Notes to 2016.11.8 Release Notes

garethgreenaway committed:
Merge pull request #43776 from Ch3LL/2016.11.8_docs

[2016.11] Bump latest and previous versions

Ch3LL committed:
Add Security Notes to 2016.3.8 Release Notes

Commit:
Merge pull request #43968 from rossengeorgiev/fix-zenoss-prod_state

fix zenoss state module not respecting test=true

garethgreenaway committed:
Merge pull request #43991 from Ch3LL/3.8_sec_2

Add Security Notes to 2016.3.8 Release Notes

Commit:
Merge pull request #43960 from cro/ldap_nopw_bind2

Require that bindpw be non-empty when auth.ldap.anonymous is False

Commit:
Merge branch '2016.11' into 2016.11-fix-2291

msummers42 committed:
addresses issue #43307, disk.format_ to disk.format

This change fixes breakage. It appears the disk.format_ func is
aliased to disk.format in modules/disk.py

Ch3LL committed:
fix 2016.3.7 release notes merge conflict

Ch3LL committed:
fix 2016.3.7 release notes merge conflict

Commit:
Merge pull request #43962 from bobrik/kmod-built-in

Report built-in modiles in kmod.available, fixes #43945

benediktwerner committed:
Added missing tutorial docs to the tutorial index

benediktwerner committed:
Fixed spelling mistake in salt_bootstrap tutorial

Commit:
Merge pull request #43955 from meaksh/2016.11-fix-2291

Enable a new '--with-salt-version' parameter for the "setup.py" script

whiteinge committed:
Also catch cpstats AttributeError for bad CherryPy release ~5.6.0

Ch3LL committed:
Add 2016.11.9 Release Note File

Commit:
Merge pull request #44019 from benediktwerner/2016.11

Added missing docs to the tutorial index and fixed  spelling mistake

Commit:
Merge pull request #44023 from Ch3LL/11.9rn

Add 2016.11.9 Release Note File

dayid committed:
Typo correction of lover to lower

Commit:
Merge pull request #44011 from Ch3LL/2016.11.7_follow_up

Security Fixes for 2016.11.8

Commit:
Merge pull request #44010 from Ch3LL/2016.3.7_follow_up

Security Fixes for 2016.3.8

dayid committed:
Merge branch '2016.11' of https://github.com/saltstack/salt into lover_typo

Commit:
Merge branch '2016.3' into '2016.11'

No conflicts.

Commit:
Merge pull request #44030 from rallytime/merge-2016.11

[2016.11] Merge forward from 2016.3 to 2016.11

Commit:
Merge pull request #44028 from rallytime/bp-44011

Back-port #44011 to 2016.11.8

Commit:
Merge pull request #44025 from dayid/lover_typo

Typo correction of lover to lower

Commit:
Merge pull request #44021 from whiteinge/cpstats-attribute-error

Also catch cpstats AttributeError for bad CherryPy release ~5.6.0

gtmanfred committed:
run tests with kitchen

cetanu committed:
Catch on empty Virtualbox network addr #43427

techhat committed:
Made sure that unicoded data is sent to sha256()

techhat committed:
Use system encoding

gtmanfred committed:
don't filter if return is not a dict

techhat committed:
Apparently __salt_system_encoding__ is a thing

techhat committed:
One more encoding

gtmanfred committed:
fix pylint

gtmanfred committed:
OpenNebula does not require the template_id to be specified

garethgreenaway committed:
Merge pull request #44092 from techhat/awsunicode

Made sure that unicoded data is sent to sha256()

gtmanfred committed:
Merge branch '2016.11' into openneb

roaldnefs committed:
Format fix code example local returner doc

roaldnefs committed:
Merge branch '2016.11' into fix-doc-local-returner

msummers42 committed:
fixed test addressing issue #43307, disk.format_ to disk.format

Commit:
Merge pull request #44110 from roaldnefs/fix-doc-local-returner

Format fix code example local returner doc

Commit:
Add note about GPG signing to PR template

gtmanfred committed:
add script for copying back artifacts

gtmanfred committed:
add opennebula to Gemfile

gtmanfred committed:
test opennebula

Commit:
Merge branch '2016.11.8' into '2016.11'

No conflicts.

Commit:
Merge pull request #44097 from gtmanfred/openneb

OpenNebula does not require the template_id to be specified

Commit:
Typo fix

Commit:
Merge pull request #44124 from rallytime/merge-2016.11

[2016.11] Merge forward from 2016.11.8 to 2016.11

Commit:
Merge pull request #44122 from cachedout/gpg_pr_template

Add note about GPG signing to PR template

Commit:
Merge pull request #44093 from gtmanfred/fix-44087

don't filter if return is not a dict

gtmanfred committed:
fix output columns

gtmanfred committed:
which vagrant should go to stderr

gtmanfred committed:
use transport if not set in state_file

gtmanfred committed:
add logging

Commit:
Merge pull request #44131 from rallytime/bp-44029

Back-port #44029 to 2016.11

gtmanfred committed:
try newest salttesting

gtmanfred committed:
add changes to test return

garethgreenaway committed:
When looping through the various pre, post, up and down commands put them into the interface dict using the right internet family variable.

Commit:
Fixing default redis.host in documentation

gtmanfred committed:
fix test

terminalmage committed:
yumpkg: Check pkgname instead of name to see if it is a kernel pkg

We should have been checking pkgname instead of name all along, but this
never bit us until 6638749 started passing `None` for the name param.

Commit:
Merge pull request #44158 from rallytime/bp-44089

Back-port #44089 to 2016.11

gtmanfred committed:
missed removing changes in the next test

Commit:
Merge pull request #44188 from terminalmage/issue44150

yumpkg: Check pkgname instead of name to see if it is a kernel pkg

Commit:
Merge pull request #43994 from oeuftete/fix-manage-runner-presence

Fix manage.present to show lost minions

Commit:
Merge pull request #43830 from rallytime/bp-43644

Back-port #43644 to 2016.11

Commit:
Merge pull request #44167 from garethgreenaway/44140_debian_ip_fixes

Fixes to modules/debian_ip

rallytime commented:
Thank you for submitting this fix @gvengel and welcome to the Salt community! 🎉 

It looks like there is a conflict here. Can you rebase you change and resolve the conflict? Or perhaps you meant to add this to a different target branch?

Commit:
Merge pull request #44205 from rallytime/bp-44177

Back-port #44177 to 2016.11

Commit:
Merge pull request #44160 from gtmanfred/directory

add changes to test return

gvengel commented:
Thanks @rallytime. Yes, I fell for the shiny green "Create PR" button, and forgot to update the target branch. Per the contributing guide, this PR is based on 2015.8, which is when the bug was introduced.

The affected code as been changed in 2016.11, so there will be conflicts merging forward. I've made a note of the difference in the PR. Let me know if there is anything else I can do to help with merging it forward.

rallytime commented:
@gvengel Thanks for following up. The `2015.8` branch is no longer accepting contributions as we are not releasing from that branch any more, nor doing merge-forwards.

Would you mind moving this and making the appropriate fix to `2016.11` instead? Sorry about the confusion there.

gtmanfred committed:
fix pylint

gtmanfred committed:
add comment to Gemfile and move copyartifacts

moved copyartifacts.py to tests/support

Also added a comment to the Gemfile that this is only used to run the testsuite with kitchen-salt

gtmanfred committed:
fix pylint comments

twangboy committed:
Fix typo

terminalmage committed:
Fix log message in salt.utils.gitfs

This needed to be using the config overlay function and not referencing
the attribute directly.

Commit:
Merge pull request #44269 from terminalmage/fix-log-message

Fix log message in salt.utils.gitfs

Commit:
Merge pull request #44268 from twangboy/win_fix_lgpo_typo

Fix typo

Commit:
Merge pull request #44259 from gtmanfred/2016.11

begin switching in kitchen-salt for running the test suite

gtmanfred committed:
use our git repo for kitchen-salt

Commit:
Merge pull request #44286 from gtmanfred/2016.11

use our git repo for kitchen-salt

gvengel committed:
Fix file.line diff formatting.

Using _splitlines_preserving_trailing_newline() strips off line
endings, mangling the unified_diff output. Switching to splitlines
with the keepends option resolves this issue, while still
preserving trailing newline.

Fixes #41474
